### DISCUSSION TITLE
Issue importing pouchdb directly via nodejs

### DISCUSSION THREAD

**Discussion by johanna-herrmann** [quasarframework/quasar#12476] [General - Components / Directives / etc]
Hello, everyone.
Sorry for bad english.

Scroll to bottom, for TLDR;

I have a problem, using PouchDB on quasar.

I am using quasar v2.

I have installed pouchdb via `yarn add pouchdb`
In src/storage/storage.js i try to import PouchDB via `import PouchDB from 'pouchdb';`.
Now, when i run `quasar dev -m electron` or `quasar dev -m android`,
I get the following error:
`export 'default' (imported as 'uuid') was not found in './dist/index.js' (possible exports: __esModule, v1, v3, v4, v5)`

My first thought was, this has to with what is said here: https://quasar.dev/quasar-cli/developing-electron-apps/electron-preload-script.
So I injected the required database using:
```js
import { contextBridge } from 'electron';

import PouchDB from 'pouchdb';
const db = new PouchDB('my_database');

contextBridge.exposeInMainWorld('electronStorage', {
  getLocal: () => { return db }
})
```
And then in storage.js i simply use `const db = window.electronStorage.getLocal()`;
This workes like a charm, without any error or warning.

My problem is, i also need a PouchDB in Cordova context (src-cordova), but there is no such preload script like for electron.

My current approuch is the following (example for pouchdb):
I copy the file node_modules/pouchdb/dist/pouchdb.min.js to src/unpkg/pouchdb.js.
In src/storage/storage.js i just do `import PouchDB from '../unpkg/pouchdb.js'`
The copy step is done via postinstall script, each time after execution of `yarn`.
This approach works on desktop and mobile without errors.

But I would prefer, to use PouchDB directly from nodejs rather than use copied files.
I don't know, what exactly is the problem and why it is not working.
- Security Reasons (https://quasar.dev/quasar-cli/developing-electron-apps/electron-preload-script)?
   -> What would be the cordova pendant to electron preload script?
- A bug?
   -> At the moment i did not create a bug report, since i don't if it is a bug.

I also tried to use pouchdb directly from nodejs in https://github.com/electron-userland/electron-webpack-quick-start and it worked.
I wonder why it doesn't with quasar (i tried with newly created project (`quasar create test-app`)).
   
 TLDR;
 To questions:
 - Is there any cordova pendant to electrons preload-script?
 - Should PouchDB be usable directly from nodejs without preload script (Then my problem is a bug)?
 
 Thanks in advance for any answers.
